hashintel/hash
# Feed block externals from peer dependencies

## 🌟 What is the purpose of this PR?

This PR updates `externals` in HASH blocks. The new approach is based on picking `peerDependencies` from `package.json` instead of hard-coding `externals` in the webpack plugin.





## 🐾 Next steps

Update and merge https://github.com/blockprotocol/blockprotocol/pull/258

## 🔍 What does this change?

The new value of `externals` in `block-metada.json` matches Block Protocol spec.

Before: `{ "package-name": "package-name" }`
After: `{ "package-name": "version" }`

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1200211978612931/1201865120407905/f) (_internal_)

## 🛡 What tests cover this?

Vercel deployment in https://github.com/blockprotocol/blockprotocol/pull/258

## ❓ How to test this?

1. Browse block hub in https://github.com/blockprotocol/blockprotocol/pull/258
2. Open path like `/blocks/@hash/callout/block-metadata.json` and check the contents of `externals`. It should be something like:

    ```
    "externals": {
      "react": "^17.0.2",
      "react-dom": "^17.0.2",
      "twind": "^0.16.16"
    },
    ```
